--- a/xla/ffi/attribute_map.cc
+++ b/xla/ffi/attribute_map.cc
@@ -40,6 +40,10 @@
 #include "xla/tsl/platform/statusor.h"
 
 namespace xla::ffi {
+
+template <typename...>
+constexpr bool always_false = false;
+
 template <typename OutputVariant, typename... InputTypes>
 static OutputVariant Convert(std::variant<InputTypes...> input) {
   return std::visit(
@@ -287,7 +291,7 @@
         } else if constexpr (std::is_same_v<U, AttributesDictionary>) {
           *proto.mutable_dict() = value.ToProto();
         } else {
-          static_assert(false, "Unsupported attribute type");
+          static_assert(always_false<U>, "Unsupported attribute type");
         }
       },
       AsVariant());
@@ -322,7 +326,7 @@
         } else if constexpr (std::is_same_v<U, double>) {
           proto.set_f64(value);
         } else {
-          static_assert(false, "Unsupported scalar type");
+          static_assert(always_false<U>, "Unsupported scalar type");
         }
       },
       AsVariant());
@@ -410,7 +414,7 @@
           proto.mutable_f64()->mutable_values()->Assign(value.begin(),
                                                         value.end());
         } else {
-          static_assert(false, "Unsupported array type");
+          static_assert(always_false<U>, "Unsupported array type");
         }
       },
       AsVariant());
@@ -482,7 +486,7 @@
         } else if constexpr (std::is_same_v<U, std::string>) {
           proto.set_str(value);
         } else {
-          static_assert(false, "Unsupported flat attribute type");
+          static_assert(always_false<U>, "Unsupported flat attribute type");
         }
       },
       AsVariant());
